modules/objdetect/test/test_qrcode_decode_encode.cpp

```diff
@@ -0,0 +1,222 @@
+    ASSERT_TRUE(file_config.isOpened()) << "Can't read validation data: " << dataset_config;
+    {
+        FileNode images_list = file_config["test_images"];
+        size_t images_count = static_cast<size_t>(images_list.size());
+        ASSERT_GT(images_count, 0u) << "Can't find validation data entries in 'test_images': " << dataset_config;
+
+        for (size_t index = 0; index < images_count; index++)
+        {
+            FileNode config = images_list[(int)index];
+            std::string name_test_image = config["image_name"];
+            if (name_test_image == name_current_image)
+            {
+
+                std::string original_info = config["info"];
+                EXPECT_EQ(decoded_info, original_info);
+
+                return; // done
+            }
+        }
+        std::cerr
+            << "Not found results for '" << name_current_image
+            << "' image in config file:" << dataset_config << std::endl
+            << "Re-run tests with enabled UPDATE_DECODE_TEST_DATA macro to update test data."
+            << std::endl;
+    }
+}
+
+        typedef testing::TestWithParam< std::string > Objdetect_QRCode_Encode;
+        TEST_P(Objdetect_QRCode_Encode, regression){
+            const double ecc_capacity[4] = {0.07,0.15,0.25,0.30};
+
+            const std::string name_current_image = GetParam();
+            const std::string root = "qrcode/encode";
+
+            std::string image_path = findDataFile(root + name_current_image);
+            /**read the original image */
+            Mat src = imread(image_path, IMREAD_GRAYSCALE), straight_barcode;
+            ASSERT_FALSE(src.empty()) << "Can't read image: " << image_path;
+
+            std::vector<Point> corners;
+            std::string decoded_info;
+
+            const std::string dataset_config = findDataFile(root + "dataset_config.json");
+            FileStorage file_config(dataset_config, FileStorage::READ);
+            ASSERT_TRUE(file_config.isOpened()) << "Can't read validation data: " << dataset_config;
+            {
+                FileNode images_list = file_config["test_images"];
+                size_t images_count = static_cast<size_t>(images_list.size());
+                ASSERT_GT(images_count, 0u) << "Can't find validation data entries in 'test_images': " << dataset_config;
+
+                for (size_t index = 0; index < images_count; index++)
+                {
+                    FileNode config = images_list[(int)index];
+                    std::string name_test_image = config["image_name"];
+
+                    if (name_test_image == name_current_image)
+                    {
+                        int mode = config["mode"];
+                        int version = config["version"];; 
+                        int ecc_level = config["ecc_level"];;
+                        int mask_type = config["mask_type"];;
+                        int eci_num = config["eci_num"];;
```

Comment:
As we said above, in this test we will only check the work of the auto mode, so we pass only string in `QRCodeEncoder::generate`, others parameters are by default. 
And once again, we get information about string and image name from file generated for decoding